pkg/beacon/dkg/result/submission.go

```diff
@@ -79,26 +79,19 @@ func (sm *SubmittingMember) SubmitDKGResult(
 			onSubmittedResultChan <- event.BlockNumber
 		},
 	)
-
-	returnWithError := func(err error) error {
-		subscription.Unsubscribe()
-		close(onSubmittedResultChan)
-		return err
-	}
+	defer subscription.Unsubscribe()
 
 	alreadySubmitted, err := chainRelay.IsGroupRegistered(result.GroupPublicKey)
 	if err != nil {
-		return returnWithError(
-			fmt.Errorf(
-				"could not check if the result is already submitted: [%v]",
-				err,
-			),
+		return fmt.Errorf(
+			"could not check if the result is already submitted: [%v]",
```

Comment:
Maybe we could use [%w] instead of [%v] for nested errors in the whole file?

Reply:
See https://github.com/keep-network/keep-core/pull/3196/commits/be83c97dfc2c654a46247c322026ef0dce0d9013